behaviour/graql/language/delete.feature

```diff
@@ -604,52 +978,327 @@ Feature: Graql Delete Query
       """
     Then uniquely identify answer concepts
-      | n     |
-      | lnST  |
-
-    Then get answers of graql query
+      | n    |
+      | lnST |
+    When get answers of graql query
       """
       match $x isa person, has lastname $n; get;
       """
     Then uniquely identify answer concepts
-      | x     | n      |
-      | JOHN  | lnST   |
+      | x    | n    |
+      | JOHN | lnST |
+
+
+  Scenario: an attribute ownership can be deleted using the 'attribute' meta label
+    Given graql define
+      """
+      define
+      address sub attribute, value string;
+      postcode sub address;
+      person has postcode;
+      """
+    Given the integrity is validated
+    Given get answers of graql insert
+      """
+      insert
+      $x isa person, has name "Sherlock", has postcode "W1U8ED";
+      """
+    Given the integrity is validated
+    Given concept identifiers are
+      |      | check | value           |
+      | SHER | key   | name:Sherlock   |
+      | nSLK | value | name:Sherlock   |
+      | pcW1 | value | postcode:W1U8ED |
+    Then uniquely identify answer concepts
+      | x    |
+      | SHER |
+    When get answers of graql query
+      """
+      match $x has attribute $a; get;
+      """
+    Then uniquely identify answer concepts
+      | x    | a    |
+      | SHER | nSLK |
+      | SHER | pcW1 |
+    When graql delete
+      """
+      match
+        $x isa person, has attribute $a;
+        $a isa postcode;
+      delete
+        $x has attribute $a;
+      """
+    Then the integrity is validated
+    When get answers of graql query
+      """
+      match $x has attribute $a; get;
+      """
+    Then uniquely identify answer concepts
+      | x    | a    |
+      | SHER | nSLK |
+
+
+  Scenario: an attribute ownership can be deleted using its supertype as a label
+    Given graql define
+      """
+      define
+      address sub attribute, value string;
+      postcode sub address;
+      person has postcode;
+      """
+    Given the integrity is validated
+    Given graql insert
+      """
+      insert
+      $x isa person, has name "Sherlock", has postcode "W1U8ED";
+      """
+    Given the integrity is validated
+    Given concept identifiers are
+      |      | check | value           |
+      | SHER | key   | name:Sherlock   |
+      | nSLK | value | name:Sherlock   |
+      | pcW1 | value | postcode:W1U8ED |
+    When get answers of graql query
+      """
+      match $x has address $a; get;
+      """
+    Then uniquely identify answer concepts
+      | x    | a    |
+      | SHER | pcW1 |
+    When graql delete
+      """
+      match
+        $x isa person, has address $a;
+      delete
+        $x has address $a;
+      """
+    Then the integrity is validated
+    When get answers of graql query
+      """
+      match $x has address $a; get;
+      """
+    Then answer size is: 0
+
+
+  Scenario: delete an attribute ownership using too-specific (downcasting) type throws
```

Comment:
note that this and the following scenarios use a different case ("delete" vs "deleting"), we should settle on one

Reply:
Aaaaaaah.